hippoclient/product.py

```diff
@@ -352,3 +360,40 @@ def uncache(client: Client, cache: MultiCache, id: str) -> None:
                 console.print(f"Removed file {source.name} ({source.uuid}) from cache")
 
     return
+
+
+def set_visibility(client: Client, id: str, visibility: str) -> None:
+    """
+    Update the visibility of a product.
+
+    Arguments
+    ---------
+    client: Client
+        The client to use for interacting with the hippo API.
+    collection_id : str
+        The ID of the product whose visibility is being updated.
+    visibility : str
+        The new visibility level ('public', 'collaboration', or 'private').
+
+    Raises
+    ------
+    ValueError
+        If an invalid visibility level is provided.
+    """
+    # Validate the provided visibility level
+    try:
+        visibility_enum = Visibility(visibility)  # Convert string to Visibility enum
+    except ValueError:
+        raise ValueError(
+            "Invalid visibility level. Choose from 'public', 'collaboration', or 'private'."
+        )
```

Comment:
I don't like this repetition everywhere in the code; your Visbility enum should raise an appropriate error